### ISSUE TITLE
You are likely vulnerable to Unicode overlong character sequences

### ISSUE THREAD

**Issue by cameronkerrnz** [revpoint/logstash-codec-avro_schema_registry#25]
Hello, I maintain a repository that I forked from this repository (because I needed to add some local concerns that wouldn't be appropriate to feed back to this repository). Recently I found that my pipeline would occassionally crash because of web access logs that contained attempts at path traversal using overlong character sequences.

Because the input's encoder didn't sanitize for this condition (for example by replacing illegal character sequences with a Unicode replacement character, which is what the standard codecs seem to do) it caused an unhandled exception when Logstash evaluated a conditional such as

```
if [message] =~ /foo/ { ... }
```

This caused the pipeline to shutdown. The error in the log (using Logstash 7.14.0) would appear like this:

```
[2021-08-11T13:04:18,744][ERROR][logstash.javapipeline    ][main] Pipeline worker error, the pipeline will be stopped {:pipeline_id=>"main", :error=>"(ArgumentError) invalid byte sequence in UTF-8", :exception=>Java::OrgJrubyExceptions::ArgumentError, :backtrace=>["org.jruby.RubyRegexp.match?(org/jruby/RubyRegexp.java:1180)", "usr.share.logstash.logstash_minus_core.lib.logstash.java_pipeline.start_workers(/usr/share/logstash/logstash-core/lib/logstash/java_pipeline.rb:295)"], :thread=>"#<Thread:0x2230aabb sleep>"}
[2021-08-11T13:04:18,749][INFO ][filewatch.observingread  ][main] QUIT - closing all files and shutting down.
[2021-08-11T13:04:18,750][WARN ][logstash.javapipeline    ][main] Waiting for input plugin to close {:pipeline_id=>"main", :thread=>"#<Thread:0x2230aabb run>"}
```

I have a post at https://discuss.elastic.co/t/logstash-7-14-0-invalid-byte-sequence-in-utf-8-in-logstash-javapipeline/280976/5

The short version is that for each string field you should use a LogStash::Util::Charset object and .convert it; this uses `String.valid_encoding?` to detect if a string is dodgy and then does a hack to clean it up a bit.
